### PR TITLE
Ports: Ghost Pool Protection Admin Button

### PR DESCRIPTION
**Pull request by Crossedfall**
Ports https://github.com/tgstation/tgstation/pull/52683

-----

## About The Pull Request

You can disable certain ways ghosts can get back into the round to keep a living ghost pool for events to pull from. There is no option to prevent in round revives as that's not what this is here to prevent,, this should only be for ghosts creating new bodies to leave the pool, thus putting themselves out of the event.
## I JUST DISABLED EVERYTHING! MUAHAHAHA!
image

## Why It's Good For The Game

Admins need ghosts sometimes for events, so being able to cut off certain ghost sinks will help us out. I fucking hate running out of ghosts.

## Changelog
:cl:tralezab
admin: New Adminbus button- Ghost Pool Protection! It lets you enable and disable which sources ghosts can rejoin the round, minus their physical bodies getting revived.
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
Briefly tested it and it all seemed fine. Could go through a TM if desired though

**Comment by francinum** [comment]
UI buttons are uncomfortably stuffed together for some reason?

**Comment by Crossedfall** [comment]
Fixed the UI (thanks @MCterra10)

image

**Review by francinum — APPROVED** [review]
Seems to function fine in testing.

File: code/modules/events/_event.dm [hunk 1]
```diff
@@ -53,6 +53,8 @@
 		return FALSE
 	if(holidayID && (!SSevents.holidays || !SSevents.holidays[holidayID]))
 		return FALSE
+	if(ispath(typepath, /datum/round_event/ghost_role) && GHOSTROLE_MIDROUND_EVENT)
+		return FALSE
```
Comment (PowerfulBacon) [inline code/modules/events/_event.dm:57]: This broke midround spawning and hasn't been noticed until now dogkek

**Pull request merged**